### ISSUE TITLE
NotImplementedError when following instructions in CONTRIBUTING.md

### ISSUE THREAD

**Issue by rain1** [mitmproxy/mitmproxy#4791]
#### Problem Description
Followed instructions in CONTRIBUTING.md. When I run mitmproxy I get error NotImplementedError. When I run mitmweb when it works just fine. It even managed to intercept https requests. I suspect there is somethign wrong with command line version of it.

#### Steps to reproduce the behavior:
1. install mitmproxy from this repo under windows with Python 3.9.7
2. run venv\Scripts\activate
3. run mitmproxy

#### System Information
(venv) C:\projektid\mitmproxy-main>mitmproxy --version
Mitmproxy: 8.0.0.dev
Python:    3.9.7
OpenSSL:   OpenSSL 1.1.1l  24 Aug 2021
Platform:  Windows-10-10.0.17134-SP0

Full cmd output:

```
Microsoft Windows [Version 10.0.17134.1246]
(c) 2018 Microsoft Corporation. All rights reserved.

C:\projektid\mitmproxy-main>python --version
Python 3.9.7

C:\projektid\mitmproxy-main>ls
CHANGELOG.md  codecov.yml  CONTRIBUTING.md  docs  examples  LICENSE  MANIFEST.in  mitmproxy  README.md  release  setup.cfg  setup.py  test  tox.ini  web

C:\projektid\mitmproxy-main>python -m venv venv

C:\projektid\mitmproxy-main>venv\Scripts\pip install -e .[dev]
Obtaining file:///C:/projektid/mitmproxy-main
Collecting asgiref<3.5,>=3.2.10
  Downloading asgiref-3.4.1-py3-none-any.whl (25 kB)
Collecting blinker<1.5,>=1.4
  Downloading blinker-1.4.tar.gz (111 kB)
     |████████████████████████████████| 111 kB 6.4 MB/s
Collecting Brotli<1.1,>=1.0
  Downloading Brotli-1.0.9-cp39-cp39-win_amd64.whl (383 kB)
     |████████████████████████████████| 383 kB 6.4 MB/s
Collecting certifi>=2019.9.11
  Downloading certifi-2021.5.30-py2.py3-none-any.whl (145 kB)
     |████████████████████████████████| 145 kB ...
Collecting click<8.1,>=7.0
  Downloading click-8.0.1-py3-none-any.whl (97 kB)
     |████████████████████████████████| 97 kB ...
Collecting cryptography<3.5,>=3.3
  Downloading cryptography-3.4.8-cp36-abi3-win_amd64.whl (1.6 MB)
     |████████████████████████████████| 1.6 MB 6.4 MB/s
Collecting flask<2.1,>=1.1.1
  Downloading Flask-2.0.1-py3-none-any.whl (94 kB)
     |████████████████████████████████| 94 kB ...
Collecting h11<0.13,>=0.11
  Downloading h11-0.12.0-py3-none-any.whl (54 kB)
     |████████████████████████████████| 54 kB ...
Collecting h2<5,>=4.0
  Downloading h2-4.0.0-py3-none-any.whl (57 kB)
     |████████████████████████████████| 57 kB ...
Collecting hyperframe<7,>=6.0
  Downloading hyperframe-6.0.1-py3-none-any.whl (12 kB)
Collecting kaitaistruct<0.10,>=0.7
  Downloading kaitaistruct-0.9.tar.gz (5.5 kB)
Collecting ldap3<2.10,>=2.8
  Downloading ldap3-2.9.1-py2.py3-none-any.whl (432 kB)
     |████████████████████████████████| 432 kB 6.4 MB/s
Collecting msgpack<1.1.0,>=1.0.0
  Downloading msgpack-1.0.2-cp39-cp39-win_amd64.whl (68 kB)
     |████████████████████████████████| 68 kB ...
Collecting passlib<1.8,>=1.6.5
  Downloading passlib-1.7.4-py2.py3-none-any.whl (525 kB)
     |████████████████████████████████| 525 kB ...
Collecting protobuf<3.18,>=3.14
  Downloading protobuf-3.17.3-cp39-cp39-win_amd64.whl (909 kB)
     |████████████████████████████████| 909 kB ...
Collecting pyOpenSSL<20.1,>=20.0
  Downloading pyOpenSSL-20.0.1-py2.py3-none-any.whl (54 kB)
     |████████████████████████████████| 54 kB ...
Collecting pyparsing<2.5,>=2.4.2
  Downloading pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
     |████████████████████████████████| 67 kB ...
Collecting pyperclip<1.9,>=1.6.0
  Downloading pyperclip-1.8.2.tar.gz (20 kB)
Collecting ruamel.yaml<0.17.14,>=0.16
  Downloading ruamel.yaml-0.17.13-py3-none-any.whl (108 kB)
     |████████████████████████████████| 108 kB ...
Collecting sortedcontainers<2.5,>=2.3
  Downloading sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)
Collecting tornado<7,>=4.3
  Downloading tornado-6.1-cp39-cp39-win_amd64.whl (422 kB)
     |████████████████████████████████| 422 kB ...
Collecting urwid<2.2,>=2.1.1
  Downloading urwid-2.1.2.tar.gz (634 kB)
     |████████████████████████████████| 634 kB ...
Collecting wsproto<1.1,>=1.0
  Downloading wsproto-1.0.0-py3-none-any.whl (24 kB)
Collecting publicsuffix2<3,>=2.20190812
  Downloading publicsuffix2-2.20191221-py2.py3-none-any.whl (89 kB)
     |████████████████████████████████| 89 kB 6.1 MB/s
Collecting zstandard<0.16,>=0.11
  Downloading zstandard-0.15.2-cp39-cp39-win_amd64.whl (582 kB)
     |████████████████████████████████| 582 kB 6.4 MB/s
Collecting pydivert<2.2,>=2.0.3
  Downloading pydivert-2.1.0-py2.py3-none-any.whl (104 kB)
     |████████████████████████████████| 104 kB ...
Collecting hypothesis<7,>=5.8
  Downloading hypothesis-6.17.4-py3-none-any.whl (371 kB)
     |████████████████████████████████| 371 kB 6.4 MB/s
Collecting parver<2.0,>=0.1
  Downloading parver-0.3.1-py2.py3-none-any.whl (14 kB)
Collecting pdoc>=4.0.0
  Downloading pdoc-7.4.0-py3-none-any.whl (106 kB)
     |████████████████████████████████| 106 kB 6.4 MB/s
Collecting pyinstaller==4.5.1
  Downloading pyinstaller-4.5.1-py3-none-win_amd64.whl (1.9 MB)
     |████████████████████████████████| 1.9 MB ...
Collecting pytest-asyncio!=0.14,<0.16,>=0.10.0
  Downloading pytest_asyncio-0.15.1-py3-none-any.whl (12 kB)
Collecting pytest-cov<3,>=2.7.1
  Downloading pytest_cov-2.12.1-py2.py3-none-any.whl (20 kB)
Collecting pytest-timeout<2,>=1.3.3
  Downloading pytest_timeout-1.4.2-py2.py3-none-any.whl (10 kB)
Collecting pytest-xdist<3,>=2.1.0
  Downloading pytest_xdist-2.3.0-py3-none-any.whl (37 kB)
Collecting pytest<7,>=6.1.0
  Downloading pytest-6.2.5-py3-none-any.whl (280 kB)
     |████████████████████████████████| 280 kB ...
Collecting requests<3,>=2.9.1
  Downloading requests-2.26.0-py2.py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB ...
Collecting tox<4,>=3.5
  Downloading tox-3.24.3-py2.py3-none-any.whl (85 kB)
     |████████████████████████████████| 85 kB 6.1 MB/s
Collecting wheel<0.38,>=0.36.2
  Downloading wheel-0.37.0-py2.py3-none-any.whl (35 kB)
Collecting pyinstaller-hooks-contrib>=2020.6
  Downloading pyinstaller_hooks_contrib-2021.3-py2.py3-none-any.whl (200 kB)
     |████████████████████████████████| 200 kB ...
Collecting altgraph
  Downloading altgraph-0.17-py2.py3-none-any.whl (21 kB)
Collecting pefile>=2017.8.1
  Downloading pefile-2021.9.3.tar.gz (72 kB)
     |████████████████████████████████| 72 kB 249 kB/s
Collecting pywin32-ctypes>=0.2.0
  Downloading pywin32_ctypes-0.2.0-py2.py3-none-any.whl (28 kB)
Requirement already satisfied: setuptools in c:\projektid\mitmproxy-main\venv\lib\site-packages (from pyinstaller==4.5.1->mitmproxy==8.0.0.dev0) (57.4.0)
Collecting colorama
  Downloading colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Collecting cffi>=1.12
  Downloading cffi-1.14.6-cp39-cp39-win_amd64.whl (180 kB)
     |████████████████████████████████| 180 kB 6.4 MB/s
Collecting pycparser
  Downloading pycparser-2.20-py2.py3-none-any.whl (112 kB)
     |████████████████████████████████| 112 kB ...
Collecting Jinja2>=3.0
  Downloading Jinja2-3.0.1-py3-none-any.whl (133 kB)
     |████████████████████████████████| 133 kB 6.4 MB/s
Collecting itsdangerous>=2.0
  Downloading itsdangerous-2.0.1-py3-none-any.whl (18 kB)
Collecting Werkzeug>=2.0
  Downloading Werkzeug-2.0.1-py3-none-any.whl (288 kB)
     |████████████████████████████████| 288 kB 6.8 MB/s
Collecting hpack<5,>=4.0
  Downloading hpack-4.0.0-py3-none-any.whl (32 kB)
Collecting attrs>=19.2.0
  Downloading attrs-21.2.0-py2.py3-none-any.whl (53 kB)
     |████████████████████████████████| 53 kB 943 kB/s
Collecting MarkupSafe>=2.0
  Downloading MarkupSafe-2.0.1-cp39-cp39-win_amd64.whl (14 kB)
Collecting pyasn1>=0.4.6
  Downloading pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
     |████████████████████████████████| 77 kB ...
Collecting six~=1.13
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting arpeggio~=1.7
  Downloading Arpeggio-1.10.2-py2.py3-none-any.whl (54 kB)
     |████████████████████████████████| 54 kB 3.8 MB/s
Collecting pygments
  Downloading Pygments-2.10.0-py3-none-any.whl (1.0 MB)
     |████████████████████████████████| 1.0 MB 6.4 MB/s
Collecting future
  Downloading future-0.18.2.tar.gz (829 kB)
     |████████████████████████████████| 829 kB ...
Collecting packaging
  Downloading packaging-21.0-py3-none-any.whl (40 kB)
     |████████████████████████████████| 40 kB ...
Collecting pluggy<2.0,>=0.12
  Downloading pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting iniconfig
  Downloading iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Collecting atomicwrites>=1.0
  Downloading atomicwrites-1.4.0-py2.py3-none-any.whl (6.8 kB)
Collecting toml
  Downloading toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting py>=1.8.2
  Downloading py-1.10.0-py2.py3-none-any.whl (97 kB)
     |████████████████████████████████| 97 kB ...
Collecting coverage>=5.2.1
  Downloading coverage-5.5-cp39-cp39-win_amd64.whl (211 kB)
     |████████████████████████████████| 211 kB ...
Collecting execnet>=1.1
  Downloading execnet-1.9.0-py2.py3-none-any.whl (39 kB)
Collecting pytest-forked
  Downloading pytest_forked-1.3.0-py2.py3-none-any.whl (4.7 kB)
Collecting idna<4,>=2.5
  Downloading idna-3.2-py3-none-any.whl (59 kB)
     |████████████████████████████████| 59 kB ...
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.6-py2.py3-none-any.whl (138 kB)
     |████████████████████████████████| 138 kB 6.8 MB/s
Collecting charset-normalizer~=2.0.0
  Downloading charset_normalizer-2.0.4-py3-none-any.whl (36 kB)
Collecting ruamel.yaml.clib>=0.1.2
  Downloading ruamel.yaml.clib-0.2.6-cp39-cp39-win_amd64.whl (118 kB)
     |████████████████████████████████| 118 kB ...
Collecting filelock>=3.0.0
  Downloading filelock-3.0.12-py3-none-any.whl (7.6 kB)
Collecting virtualenv!=20.0.0,!=20.0.1,!=20.0.2,!=20.0.3,!=20.0.4,!=20.0.5,!=20.0.6,!=20.0.7,>=16.0.0
  Downloading virtualenv-20.7.2-py2.py3-none-any.whl (5.3 MB)
     |████████████████████████████████| 5.3 MB ...
Collecting platformdirs<3,>=2
  Downloading platformdirs-2.3.0-py3-none-any.whl (13 kB)
Collecting distlib<1,>=0.3.1
  Downloading distlib-0.3.2-py2.py3-none-any.whl (338 kB)
     |████████████████████████████████| 338 kB ...
Collecting backports.entry-points-selectable>=1.0.4
  Downloading backports.entry_points_selectable-1.1.0-py2.py3-none-any.whl (6.2 kB)
Using legacy 'setup.py install' for blinker, since package 'wheel' is not installed.
Using legacy 'setup.py install' for kaitaistruct, since package 'wheel' is not installed.
Using legacy 'setup.py install' for pefile, since package 'wheel' is not installed.
Using legacy 'setup.py install' for pyperclip, since package 'wheel' is not installed.
Using legacy 'setup.py install' for urwid, since package 'wheel' is not installed.
Using legacy 'setup.py install' for future, since package 'wheel' is not installed.
Installing collected packages: pyparsing, pycparser, toml, py, pluggy, packaging, MarkupSafe, iniconfig, colorama, cffi, attrs, atomicwrites, Werkzeug, six, ruamel.yaml.clib, pytest, pyasn1, platformdirs, Jinja2, itsdangerous, hyperframe, hpack, h11, future, filelock, distlib, cryptography, click, backports.entry-points-selectable, zstandard, wsproto, virtualenv, urwid, urllib3, tornado, sortedcontainers, ruamel.yaml, pywin32-ctypes, pytest-forked, pyperclip, pyOpenSSL, pyinstaller-hooks-contrib, pygments, pydivert, publicsuffix2, protobuf, pefile, passlib, msgpack, ldap3, kaitaistruct, idna, h2, flask, execnet, coverage, charset-normalizer, certifi, Brotli, blinker, asgiref, arpeggio, altgraph, wheel, tox, requests, pytest-xdist, pytest-timeout, pytest-cov, pytest-asyncio, pyinstaller, pdoc, parver, mitmproxy, hypothesis
    Running setup.py install for future ... done
    Running setup.py install for urwid ... done
    Running setup.py install for pyperclip ... done
    Running setup.py install for pefile ... done
    Running setup.py install for kaitaistruct ... done
    Running setup.py install for blinker ... done
  Running setup.py develop for mitmproxy
Successfully installed Brotli-1.0.9 Jinja2-3.0.1 MarkupSafe-2.0.1 Werkzeug-2.0.1 altgraph-0.17 arpeggio-1.10.2 asgiref-3.4.1 atomicwrites-1.4.0 attrs-21.2.0 backports.entry-points-selectable-1.1.0 blinker-1.4 certifi-2021.5.30 cffi-1.14.6 charset-normalizer-2.0.4 click-8.0.1 colorama-0.4.4 coverage-5.5 cryptography-3.4.8 distlib-0.3.2 execnet-1.9.0 filelock-3.0.12 flask-2.0.1 future-0.18.2 h11-0.12.0 h2-4.0.0 hpack-4.0.0 hyperframe-6.0.1 hypothesis-6.17.4 idna-3.2 iniconfig-1.1.1 itsdangerous-2.0.1 kaitaistruct-0.9 ldap3-2.9.1 mitmproxy-8.0.0.dev0 msgpack-1.0.2 packaging-21.0 parver-0.3.1 passlib-1.7.4 pdoc-7.4.0 pefile-2021.9.3 platformdirs-2.3.0 pluggy-1.0.0 protobuf-3.17.3 publicsuffix2-2.20191221 py-1.10.0 pyOpenSSL-20.0.1 pyasn1-0.4.8 pycparser-2.20 pydivert-2.1.0 pygments-2.10.0
pyinstaller-4.5.1 pyinstaller-hooks-contrib-2021.3 pyparsing-2.4.7 pyperclip-1.8.2 pytest-6.2.5 pytest-asyncio-0.15.1 pytest-cov-2.12.1 pytest-forked-1.3.0 pytest-timeout-1.4.2 pytest-xdist-2.3.0 pywin32-ctypes-0.2.0 requests-2.26.0 ruamel.yaml-0.17.13 ruamel.yaml.clib-0.2.6 six-1.16.0 sortedcontainers-2.4.0 toml-0.10.2 tornado-6.1 tox-3.24.3 urllib3-1.26.6 urwid-2.1.2 virtualenv-20.7.2 wheel-0.37.0 wsproto-1.0.0 zstandard-0.15.2
WARNING: You are using pip version 21.2.3; however, version 21.2.4 is available.
You should consider upgrading via the 'C:\projektid\mitmproxy-main\venv\Scripts\python.exe -m pip install --upgrade pip' command.

C:\projektid\mitmproxy-main>mitmdump --version
'mitmdump' is not recognized as an internal or external command,
operable program or batch file.

C:\projektid\mitmproxy-main>venv\Scripts\activate

(venv) C:\projektid\mitmproxy-main>mitmdump --version
Mitmproxy: 8.0.0.dev
Python:    3.9.7
OpenSSL:   OpenSSL 1.1.1l  24 Aug 2021
Platform:  Windows-10-10.0.17134-SP0

(venv) C:\projektid\mitmproxy-main>mitmproxy --version
Mitmproxy: 8.0.0.dev
Python:    3.9.7
OpenSSL:   OpenSSL 1.1.1l  24 Aug 2021
Platform:  Windows-10-10.0.17134-SP0

(venv) C:\projektid\mitmproxy-main>python --version
Python 3.9.7

(venv) C:\projektid\mitmproxy-main>ls
CHANGELOG.md  codecov.yml  CONTRIBUTING.md  docs  examples  LICENSE  MANIFEST.in  mitmproxy  mitmproxy.egg-info  README.md  release  setup.cfg  setup.py  test  tox.ini  venv  web

(venv) C:\projektid\mitmproxy-main>mitmproxy
Traceback (most recent call last):
  File "c:\projektid\mitmproxy-main\mitmproxy\master.py", line 54, in run_loop
    loop()
  File "C:\projektid\mitmproxy-main\venv\lib\site-packages\urwid\main_loop.py", line 287, in run
    self._run()
  File "C:\projektid\mitmproxy-main\venv\lib\site-packages\urwid\main_loop.py", line 377, in _run
    self.start()
  File "C:\projektid\mitmproxy-main\venv\lib\site-packages\urwid\main_loop.py", line 353, in start
    self._reset_input_descriptors()
  File "C:\projektid\mitmproxy-main\venv\lib\site-packages\urwid\main_loop.py", line 373, in _reset_input_descriptors
    self.screen.hook_event_loop(self.event_loop, self._update)
  File "c:\projektid\mitmproxy-main\mitmproxy\contrib\urwid\raw_display.py", line 470, in hook_event_loop
    handles = [event_loop.watch_file(fd, wrapper) for fd in fds]
  File "c:\projektid\mitmproxy-main\mitmproxy\contrib\urwid\raw_display.py", line 470, in <listcomp>
    handles = [event_loop.watch_file(fd, wrapper) for fd in fds]
  File "C:\projektid\mitmproxy-main\venv\lib\site-packages\urwid\main_loop.py", line 1432, in watch_file
    self._loop.add_reader(fd, callback)
  File "C:\Users\Rain\AppData\Local\Programs\Python\Python39\lib\asyncio\events.py", line 504, in add_reader
    raise NotImplementedError
NotImplementedError

mitmproxy has crashed!
Please lodge a bug report at:
        https://github.com/mitmproxy/mitmproxy/issues
sys:1: RuntimeWarning: coroutine 'BaseEventLoop._create_server_getaddrinfo' was never awaited
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

**Comment by mhils** [comment]
Interesting, thanks for the report. Do the 7.0.2 Windows binaries work for you?

**Comment by mhils** [comment]
Fixed with #4806, thanks again! :cake
